src/core/Form/Textarea/Textarea.tsx

```diff
@@ -1,52 +1,162 @@
+import classnames from 'classnames';
 import { TokensProp, InternalTokensProp } from '../../theme';
-import {
-  Textarea as CompTextarea,
-  TextareaProps as CompTextareaProps,
-} from '../../../components/Form/Textarea';
 import { baseStyles } from './Textarea.baseStyles';
 import { withSuomifiDefaultProps } from '../../theme/utils';
-import classnames from 'classnames';
+import { HtmlTextarea, HtmlTextareaProps, HtmlDiv } from '../../../reset';
+import { AutoId } from '../../../utils/AutoId';
+import { LabelText } from '../LabelText/LabelText';
+import { HintText } from '../HintText/HintText';
+import { StatusText } from '../StatusText/StatusText';
+import { InputStatus } from '../types';
 
 const baseClassName = 'fi-textarea';
-
 const textareaClassNames = {
+  textareaContainer: `${baseClassName}_textarea-element-container`,
+  textarea: `${baseClassName}_textarea`,
   disabled: `${baseClassName}--disabled`,
   error: `${baseClassName}--error`,
 };
 
-export interface TextareaProps extends CompTextareaProps, TokensProp {}
+type TextareaStatus = Exclude<InputStatus, 'success'>;
 
-const StyledTextarea = styled(
-  ({ tokens, ...passProps }: TextareaProps & InternalTokensProp) => (
-    <CompTextarea {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
-`;
+export interface TextareaProps extends HtmlTextareaProps, TokensProp {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on textarea text change */
+  onChange?: (event: ChangeEvent<HTMLTextAreaElement>) => void;
+  /** To execute on textarea text onBlur */
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: TextareaStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  /** Optional text that is shown after labelText. Will be wrapped in parentheses. */
+  optionalText?: string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Input name */
+  name?: string;
+}
 
-export class Textarea extends Component<TextareaProps> {
+class BaseTextarea extends Component<TextareaProps> {
```

Comment:
Should probably omit some props here? E.g. type and if something else is limited by the component structure or functionality.

Reply:
Not sure what props those could be and there was no change to that part regarding this PR, but I'm open for suggestions to improve it.

It seems that the textarea [has no](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) `type` attribute and when tested, it has therefore no effect if given.

Reply:
I guess it's not mandatory if there really is no functionality that would break if some props like readonly or some event listeners are given. This is completely possible in this case, just seemed weird that there would be nothing to omit.

Reply:
lib.dom.d.ts has a loads of properties for HTMLTextAreaElement which is the underlying type.

Reply:
Okay, I will go through those and check if causing any troubles.

Reply:
`placeholder` attribute is now omitted as we have `visualPlaceholder` prop for it.